### PR TITLE
HHH-13783 Avoid SequenceMismatchStrategy checking for non-pooled optimizer like HiLo

### PR DESCRIPTION
**Pull request by NathanQingyangXu**
https://hibernate.atlassian.net/browse/HHH-13783

straightforward fixing with testing case

### CODE REVIEW

**Comment by dreab8** [comment]
applied upstream.
Thanks @NathanQingyangXu

File: hibernate-core/src/test/java/org/hibernate/test/idgen/enhanced/sequence/HiLoSequenceMismatchStrategyTest.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+				Statement statement = connection.createStatement() ) {
+			for ( String dropSequenceStatement : dropSequenceStatements ) {
+				statement.execute( dropSequenceStatement );
+			}
+			for ( String createSequenceStatement : createSequenceStatements ) {
+				statement.execute( createSequenceStatement );
+			}
+		}
+		catch (SQLException e) {
+			fail( e.getMessage() );
```
Comment (jwgmeligmeyling) [inline hibernate-core/src/test/java/org/hibernate/test/idgen/enhanced/sequence/HiLoSequenceMismatchStrategyTest.java]: Test fails under SAP HANA: 

```
org.hibernate.test.idgen.enhanced.sequence.HiLoSequenceMismatchStrategyTest > classMethod FAILED
    org.hibernate.testing.junit4.CallbackException
        Caused by: java.lang.AssertionError at HiLoSequenceMismatchStrategyTest.java:67
```

See: https://ci.hibernate.org/blue/organizations/jenkins/hibernate-orm-master-hana-main/detail/hibernate-orm-master-hana-main/910/pipeline

Anyone on it?

@dreab8  I saw you fixed this for oracle in c5581e6 ?

Comment (dreab8) [reply hibernate-core/src/test/java/org/hibernate/test/idgen/enhanced/sequence/HiLoSequenceMismatchStrategyTest.java]: Hi @jwgmeligmeyling I saw it but not working on it at the moment.

Comment (jwgmeligmeyling) [reply hibernate-core/src/test/java/org/hibernate/test/idgen/enhanced/sequence/HiLoSequenceMismatchStrategyTest.java]: @NathanQingyangXu  Have you ever tested under SAP Hana / are you willing to look into this?

(CI on other dialects only runs after merge, so you couldn't have known this)

**Comment by NathanQingyangXu** [comment]
I didn't. Is it urgent? I might be busy today on my own job and I would
appreciate it if someone takes a look at. zero experience on those DBs.

**Pull request closed without merging**